src/math.c

```diff
@@ -189,137 +255,689 @@ u24_8 u24_8_div(u24_8 x, u24_8 y)
+        sp8 = b->s4;
+        spC = 0; // Effectively unused
+        r9 = 0;
+        r5 = 0;
+        r4 = 0;
+
+        for (i = 0; i < 64; i++) {
+            r5 <<= 1;
+            if (r4 & 0x80000000)
+                r5 |= 0x1;
+
+            r4 = (r4 << 1) & ~0x1;
+            if (r7 & 0x80000000)
+                r4 |= 0x1;
+
+            r7 <<= 1;
+            if (r6 & 0x80000000)
+                r7 |= 0x1;
+
+            r6 = (r6 << 1) & ~0x1;
+
+            if (!u32_pair_less_than(r5, r4, sp4, sp8)) {
+                temp = r4;
+                r1 = TRUE;
+                r4 -= sp8;
+                r5 -= sp4;
+
+                if (temp < r4)
+                    r5--;
+            }
+            else
+                r1 = FALSE;
+
+            spC <<= 1;
+            if (r9 & 0x80000000)
+                spC |= 0x1;
+
+            r9 = (r9 << 1) & ~0x1;
+            if (r1)
+                r9 |= 0x1;
+        }
+
+        dst->s0 = spC;
+        dst->s4 = r9;
+    }
 }
 
-UNUSED u32 sub_800A068(u32 *param_1)
+void sub_800A6D0(unkStruct_80943A8 *dst, unkStruct_80943A8 *a, unkStruct_80943A8 *b)
 {
-  u32 uVar1;
-  
-  uVar1 = ((u8)param_1[0] << 0x18) | param_1[1] >> 8;
-  if ((param_1[1] & 0x8000) != 0) {
-    uVar1++;
-  }
-  return uVar1;
+    s32 s0;
+    u32 s4;
+
+    s0 = a->s0 + b->s0;
+    s4 = a->s4 + b->s4;
+    if (s4 < a->s4)
+        s0++;
+
+    dst->s0 = s0;
+    dst->s4 = s4;
+}
+
+void sub_800A6F0(unkStruct_80943A8 *dst, unkStruct_80943A8 *a, unkStruct_80943A8 *b)
+{
+    s32 s0;
+    u32 s4;
+
+    s0 = a->s0 - b->s0;
+    s4 = a->s4 - b->s4;
+    if (s4 > a->s4)
+        s0--;
+
+    dst->s0 = s0;
+    dst->s4 = s4;
+}
+
+// Similar to sub_8009F68
+UNUSED void F48_16_Pow(unkStruct_80943A8 *dst, unkStruct_80943A8 *a, s32 b)
+{
+    unkStruct_80943A8 aa;
+    s32 bb;
+    unkStruct_80943A8 res;
+
+    aa.s0 = a->s0;
+    aa.s4 = a->s4;
+    bb = b;
+    if (bb < 0)
+        bb = -bb;
+
+    res.s0 = 0;
+    res.s4 = 0x10000;
+
+    for (; bb != 0; bb >>= 1) {
+        if (bb & 1)
+            F48_16_SMul(&res, &res, &aa);
+
+        F48_16_Square(&aa);
+    }
+
+    if (b < 0) {
+        unkStruct_80943A8 idk;
+        idk.s0 = 0;
+        idk.s4 = 0x10000;
+        F48_16_SDiv(&res, &idk, &res);
+    }
+
+    dst->s0 = res.s0;
+    dst->s4 = res.s4;
 }
+
+// Similar to sub_8009FB8
+UNUSED void sub_800A78C(unkStruct_80943A8 *dst, unkStruct_80943A8 *a, unkStruct_80943A8 *b)
+{
+    u32 temp;
+    s32 i;
+    unkStruct_80943A8 sp0;
+    unkStruct_80943A8 sp8;
+    unkStruct_80943A8 sp10;
+    unkStruct_80943A8 sp18;
+
+    sp0 = *a;
+    sp8 = *b;
+    F48_16_Abs(&sp0);
+    F48_16_Abs(&sp8);
+
+    if (sub_800A2F0(&sp0, &sp8)) {
+        sp10 = sp0;
+        sp0 = sp8;
+        sp8 = sp10;
+    }
+
+    if (!F48_16_IsZero(&sp8)) {
+        i = 0;
+        goto deez; // Fakematch cuz ya (https://decomp.me/scratch/2TCrJ)
+
+        while (i != 2) {
+            F48_16_SMul(&sp8, &sp8, &sp10);
+            i++;
```

Comment:
Could this be a for, with `SMul(...), i++` in the increment?

Reply:
Already tried it many times, you can check it https://decomp.me/scratch/2TCrJ